packages/server/test/e2e/8_read_only_fs.ts

```diff
@@ -0,0 +1,55 @@
+import * as fs from 'fs'
+import * as path from 'path'
+
+const e2e = require('../support/helpers/e2e')
+const chmodr = require('chmodr')
+
+describe('e2e project dir access', function () {
+  e2e.setup()
+  const projectDir: string = path.join('/', 'tmp', '8_read_only_fs')
+
+  beforeEach(() => {
+    if (fs.existsSync(projectDir)) {
+      return
+    }
+
+    // write out the project structure
+    // we cant sue one define by source code since we need to modify the file permissions on the dir and files
+    fs.mkdirSync(path.join(projectDir))
+
+    fs.writeFileSync(path.join(projectDir, 'cypress.json'), `{"projectId": "2pz86o"}`)
+    fs.mkdirSync(path.join(projectDir, 'cypress'))
+    fs.mkdirSync(path.join(projectDir, 'cypress', 'integration'))
+    fs.writeFileSync(path.join(projectDir, 'cypress', 'integration', 'spec.js'), `it('test',() => expect(true).to.be.true )`)
+    fs.mkdirSync(path.join(projectDir, 'cypress', 'plugins'))
+    fs.writeFileSync(path.join(projectDir, 'cypress', 'plugins', 'index.js'), `module.exports = () => {}`)
+    fs.mkdirSync(path.join(projectDir, 'cypress', 'support'))
+    fs.writeFileSync(path.join(projectDir, 'cypress', 'support', 'index.js'), `module.exports = () => {}`)
+    fs.mkdirSync(path.join(projectDir, 'cypress', 'videos'))
+
+    // setup the file/folder permissions
+    // 500 means individual read + execute
+    // the dir needs executable for debugging
+    // the main test here is that it isn't writable, which would be either 6 or 7
+    // See here for more info on these numbers http://permissions-calculator.org/
+    chmodr.sync(projectDir, 0o500)
```

Comment:
whoa, i didn't know this `0o` prefix was a thing, i am only familiar with using octals via the leading 0 (like `0500`)

Reply:
Yeah, you totally have to do it this way when running in strict mode
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Deprecated_octal

Learn something new about java script everyday